cmd/tracee/main.go

```diff
@@ -154,3 +215,57 @@ func main() {
 		logger.Fatal("app", "error", err)
 	}
 }
+
+func createEventsFromSignatures(startId events.ID, sigs []detect.Signature) {
+	id := startId
+
+	for _, s := range sigs {
+		m, err := s.GetMetadata()
+		if err != nil {
+			logger.Error("failed to load event", "error", err)
+			continue
+		}
+
+		selectedEvents, err := s.GetSelectedEvents()
+		if err != nil {
+			logger.Error("failed to load event", "error", err)
+			continue
+		}
+
+		dependencies := make([]events.ID, 0)
+
+		for _, s := range selectedEvents {
+			eventID, found := events.Definitions.GetID(s.Name)
+			if !found {
+				logger.Error("failed to load event dependency", "event", s.Name)
+				continue
+			}
+
+			dependencies = append(dependencies, eventID)
+		}
+
+		event := events.NewEventDefinition(m.EventName, []string{"rules"}, dependencies)
+
+		err = events.Definitions.Add(events.ID(id), event)
+		if err != nil {
+			logger.Error("failed to add event definition", "error", err)
+			continue
+		}
+
+		id++
+	}
+
+}
+
+func getEventNames(c *cli.Context) []string {
+	// TODO: --trace event= will become --events
+	traceArgs := c.StringSlice("trace")
+	for _, optValue := range traceArgs {
+		if strings.HasPrefix(optValue, "event=") {
```

Comment:
How it works is that first equality events are collected. If there are none(from event=... or set=...) it will pick the default set.
Then it will exclude all matching events picked up before.
So this technically won't have any problem since none of the signature events are included in the default set atm.

Since this function is practically only used to collect the signature names the name could maybe use a change, or at least a comment.

Reply:
What if you define a set and then use != ?

Reply:
Example?

Reply:
for example `set=container` `event!=container_remove` will add all container events but remove container_remove. this can potentially clash with this mechanism.
But it looks like you're using this to collect the signature events only so the function should reflect that as well as docs.